tests/framework/artifacts.py

```diff
@@ -190,3 +234,67 @@ def artifacts(self):
     def __len__(self):
         """Return the artifacts array len."""
         return len(self._artifacts)
+
+
+class SnapshotType(Enum):
+    """Supported snapshot types."""
+
+    FULL = 0
+    DIFF = 1
+
+
+class Snapshot:
+    """Manages firecracker snapshots."""
+
+    def __init__(self, mem, vmstate, disks, ssh_key):
+        """Initialize mem, vmstate, disks, key."""
+        assert mem is not None
+        assert vmstate is not None
+        assert disks is not None
+        assert ssh_key is not None
+        self._mem = mem
+        self._vmstate = vmstate
+        self._disks = disks
+        self._ssh_key = ssh_key
+
+    @classmethod
+    def from_artifacts(cls,
+                       target_folder,  # Unique path
+                       mem: Artifact,
+                       vmstate: Artifact,
+                       _disks: [DiskArtifact],
+                       ssh_key: Artifact):
+        """Construct an instance from artifacts."""
+        disk_paths = []
```

Comment:
What is the effect of this? Should it be `for disk_path in _disks`?

Reply:
This function is currently WIP, I will remove it.